app/models/report_composer.rb

```diff
@@ -145,6 +145,7 @@ def build_inputs(template, input_values)
     # process values from params (including empty hash)
     template.template_inputs.each do |input|
       val = input_values[input.id.to_s].try(:[], 'value') unless input_values.nil?
+      val ||= input.default
```

Comment:
Random note and not a full review. I don't understand the full code, but it be a boolean input where `val` is `false` but `input.default` is `true`?

Reply:
If there is no conversion hidden somewhere, then this expression (in you example) will be
```ruby
val = 'false'
val ||= 'true'
val # 'false'
```
I assume that user input is not conversed and it is saved as a `String`.

Reply:
It is not for now and there is no boolean input for now.
So it seems safe, `val = input.default if val.blank? # or val.nil?` is always safer though.

Reply:
Thanks, I've changed to `val = input.default if val.blank?`